vortex-expr/src/transform/partition.rs

```diff
@@ -273,23 +274,32 @@ impl FolderMut for StructFieldExpressionSplitter<'_> {
         };
 
         if node.as_any().downcast_ref::<Identity>().is_some() {
-            let fields = (0..self.dt_ident.names().len())
-                .map(Field::Index)
+            let fields = self
```

Comment:
Is this just cloning the field names? `self.dt_ident.names().to_vec()`

Reply:
and lifting into Field::Name, this will be removed once we remove Field from exprs